packages/start/server/server-functions/server.ts

```diff
@@ -279,6 +279,27 @@ server$.hasHandler = function (route) {
   return handlers.has(route);
 };
 
+server$.createServerResultHandler = function (handler) {
+  return (...args) =>
+    handler(...args).then((data: any) => {
+      // we need to emulate what respondWith does minus error handling.
+      if (data instanceof ResponseError) {
+        return data.clone();
+      } else if (data instanceof Response) {
+        return data;
+      } else if (
+        typeof data === "object" ||
+        typeof data === "string" ||
+        typeof data === "number" ||
+        typeof data === "boolean"
+      ) {
+        return JSON.stringify(data);
```

Comment:
This could be optimized to only do it on object but to make the check consistent with `repondWith` I left it as is